repository/src/main/java/org/pentaho/platform/repository/RepositoryFilenameUtils.java

```diff
@@ -511,7 +511,12 @@ public static String getBaseName( final String filename ) {
    * @return the extension of the file or an empty string if none exists.
    */
   public static String getExtension( final String filename ) {
-    return FilenameUtils.getExtension( filename );
+    String tempFileName = filename;
+    // PDI-19835 - FilenameUtils.getExtension() no longer supports ':' on the file path
+    if ( filename != null && filename.contains( ":" ) ) {
+      tempFileName = filename.replace( ":", "/" );
+    }
```

Comment:
The use of "String#replace(char, char)" seems to fit as a glove in this scenario.

Note that a "contains()" requires checking every character which will also be done by any "replace" invocation.
As "String#replace(char, char)" returns the same string if no replacement is needed, one could replace lines 514-518 by just:

`String tempFileName = filename.replace( ':', '/' )`

Reply:
Missing ';' in my example code ;-)

Reply:
By the way, should't it be a replaceAll()

Reply:
replaceAll uses regex :-)